open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/common/idempotency/IdempotencyValidator.java

```diff
@@ -80,123 +79,146 @@ public static IdempotencyValidationResult validateIdempotency(String idempotency
+     * @param consentId             ConsentId
+     * @return Created Time.
      */
-    protected static boolean isRequestReceivedWithinAllowedTime(long createdTime) {
-
-        if (createdTime == 0L) {
-            return false;
+    public long getCreatedTimeOfPreviousRequest(String resourcePath, String consentId) {
+        DetailedConsentResource consentRequest = null;
+        try {
+            consentRequest = consentCoreService.getDetailedConsent(consentId);
+        } catch (ConsentManagementException e) {
+            log.error(IdempotencyConstants.CONSENT_RETRIEVAL_ERROR, e);
+            return 0L;
         }
-        String allowedTimeDuration = (String) configs.get(IdempotencyConstants.IDEMPOTENCY_ALLOWED_TIME);
-        if (allowedTimeDuration != null) {
-            OffsetDateTime createdDate = OffsetDateTime.parse(convertToISO8601(createdTime));
-            OffsetDateTime currDate = OffsetDateTime.now(createdDate.getOffset());
 
-            long diffInHours = Duration.between(createdDate, currDate).toMinutes();
-            return diffInHours <= Long.parseLong(allowedTimeDuration);
-        } else {
-            log.error("Idempotency Allowed duration is null");
-            return false;
+        return consentRequest.getCreatedTime();
+    }
+
+    /**
+     * Method to get payload from previous request.
+     *
+     * @param resourcePath     Resource Path
+     * @param consentId             ConsentId
+     * @return Map containing the payload.
+     */
+    public String getPayloadOfPreviousRequest(String resourcePath, String consentId) {
+        DetailedConsentResource consentRequest = null;
+        try {
+            consentRequest = consentCoreService.getDetailedConsent(consentId);
+        } catch (ConsentManagementException e) {
+            log.error(IdempotencyConstants.CONSENT_RETRIEVAL_ERROR, e);
+            return null;
         }
+        return consentRequest.getReceipt();
     }
 
     /**
-     * Convert long date values to ISO 8601 format.
-     * @param dateValue     Date value
-     * @return ISO 8601 formatted date
+     * Method to compare whether payloads are equal.
+     *
+     * @param consentManageData   Consent Manage Data Object
+     * @param consentReceipt      Payload received from database
+     * @return   Whether payloads are equal
      */
-    public static String convertToISO8601(long dateValue) {
+    public boolean isPayloadSimilar(ConsentManageData consentManageData, String consentReceipt) {
 
-        DateFormat simple = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssXXX");
-        Date simpleDateVal = new Date(dateValue * 1000);
-        return simple.format(simpleDateVal);
+        if (consentManageData.getPayload() == null || consentReceipt == null) {
+            return false;
+        }
+
+        JsonNode expectedNode = null;
+        JsonNode actualNode = null;
+        try {
+            expectedNode = new ObjectMapper().readTree(consentManageData.getPayload().toString());
```

Comment:
Consider moving new ObjectMapper to a variable

Reply:
Fixed in https://github.com/wso2/financial-open-banking/pull/51/commits/e1205691228fc4b76853a0eab7dd99dac2516661